### ISSUE TITLE
Thought duplication after using sub categorize all and undo

### ISSUE THREAD

**Issue by shresthabijay** [cybersemics/em#967]
# Steps to reproduce

1. First create these thoughts.

```
- a
  - b
```

2. Select `a` and Subcategorize all.
3. Undo the changes.
4. Then select `a` and hit enter to create new thought below.

# Current Behavior

```
- a
  - b
- Empty Thought
  - a
    - b
```

# Expected Behavior

```
- a
  - b
- Empty Thought
```

**Comment by shresthabijay** [comment]
I came across this while I was working on #962. Seems related.

**Comment by raineorshine** [comment]
Good catch! Let's fix #962 first since this one has the compounding factor of undo. It'll be easier to narrow this down to an undo/redo bug or another `subcategorizeAll` bug.
